### PR TITLE
VACMS-6812 Create audit tool for rich text fields

### LINKED ISSUE #6812: Create an audit tool for Rich Text editors thoughout content types
## Description
We need to be able to audit content within long text, formatted fields to see if they contain unwanted markup such as headings, buttons, address blocks.. etc. 

## Acceptance Criteria
- [ ] Users can filter by content type
- [ ] Users can see the output of long text, formatted fields in the report
- [ ] Users can export the report
- [ ] ???

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by msbtterswrth**
## Description

Closes #6812 .

## Testing done
navigated to new view and followed qa steps

## Screenshots
image

## QA steps

As cms editor I can
1. navigate to /admin/content/audit/rich-text-field-audit and view a report
2. filter by content type
3. search for content within the rich text fields
4. export results to csv
- [x] Users can filter by content type
- [x] Users can see the output of long text, formatted fields in the report
- [x] Users can export the report
- [x] update url to admin/content/audit/{{name}}
- [x] update perms for content admin and admin

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [x] No announcement is needed for this code change.
  - [x] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/views.view.rich_text_field_audit.yml [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+        - 'user.node_grants:view'
+        - user.permissions
+      tags:
+        - 'config:field.storage.node.body'
+        - 'config:field.storage.node.field_additional_information_abo'
+        - 'config:field.storage.node.field_banner_alert_situationinfo'
+        - 'config:field.storage.node.field_full_story'
+        - 'config:field.storage.node.field_press_release_blurb'
+        - 'config:field.storage.node.field_press_release_fulltext'
+  page_1:
```
[the remark this answers is not in the document]
Comment (msbtterswrth) [reply config/sync/views.view.rich_text_field_audit.yml]: updated!

File: config/sync/views.view.rich_text_field_audit.yml [hunk 2]
```diff
@@ -0,0 +1,855 @@
...
+    display_plugin: page
+    id: page_1
+    display_title: Page
+    position: 1
+    display_options:
+      display_extenders: {  }
+      path: rich-text-field-audit
```
[the remark this answers is not in the document]
Comment (msbtterswrth) [reply config/sync/views.view.rich_text_field_audit.yml]: updated to /admin/content/audit/rich-text-field-audit

File: config/sync/views.view.rich_text_field_audit.yml [hunk 3]
```diff
@@ -0,0 +1,855 @@
...
+        options:
+          grouping: {  }
+          row_class: ''
+          default_row_class: true
+          override: true
+          sticky: false
+          caption: ''
+          summary: ''
+          description: ''
+          columns:
```
[the remark this answers is not in the document]
Comment (msbtterswrth) [reply config/sync/views.view.rich_text_field_audit.yml:114]: added!

File: config/sync/views.view.rich_text_field_audit.yml [hunk 4]
```diff
@@ -0,0 +1,855 @@
...
+      filters:
+        status:
+          value: '1'
+          table: node_field_data
+          field: status
+          plugin_id: boolean
+          entity_type: node
+          entity_field: status
+          id: status
+          expose:
```
[the remark this answers is not in the document]
Comment (msbtterswrth) [reply config/sync/views.view.rich_text_field_audit.yml]: also added!

**Review by swirtSJW — COMMENTED** [review]
This looking pretty good overall:
Just a couple things.  

- Update the paths in the QA steps.   So we don't need to dig through the config to figure out what and where to go.
- We have some existing patterns in terms of column order and labels.  It would be good to align yours to existing patterns
image

**Comment by msbtterswrth** [comment]
Ok i've updated the order of the view columns, I didn't know there was a specific order, but it makes me happy that there is one :) Additionally I've added the last changed date to match the view you send as a guide. 

I also fixed the link in the qa steps, thank you for catching that.

**Comment by swirtSJW** [comment]
> I didn't know there was a specific order, but it makes me happy that there is one :)
Yah it is not so much that there is a specific order as some columns will not exist on some Views, it is more about trying to have our tools have consistent feel across them.  
Things like: 

- order of columns
- naming of columns
- column sorts
- date formats
- filter order
- filter names
- edit buttons

should all provide as consistent a look and feel as possible.


Often we try to call out an existing View that is similar to model a new View upon in our refinement or planning sessions but, this one kind of bi-passed that process.

**Comment by swirtSJW** [comment]
@msbtterswrth I just pushed up a commit to address the other items in my previous comment.
image

**Review by swirtSJW — APPROVED** [review]
Approved.   Just waiting on the last test run to complete.

**Pull request merged**